### ISSUE TITLE
Bug: Avoid extraneous deps after `builder install`

### ISSUE THREAD

**Issue by exogen** [FormidableLabs/builder#15]
I think I understand why this happens, but it's still annoying. Maybe we can brainstorm a solution.

After running `builder install`, the archetype's deps get installed in your root's `node_modules`, and since they're not in your root's `package.json`, npm thinks they're extraneous. So commands like `npm ls` will print:

```
npm ERR! extraneous: babel-loader@5.3.3 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/babel-loader
npm ERR! extraneous: css-loader@0.20.2 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/css-loader
npm ERR! extraneous: file-loader@0.8.4 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/file-loader
npm ERR! extraneous: style-loader@0.12.4 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/style-loader
npm ERR! extraneous: url-loader@0.5.6 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/url-loader
npm ERR! extraneous: webpack@1.12.6 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/webpack
```

**Comment by ryan-roemer** [comment]
Hmmm.... the deps should actually be in `node_modules/ARCHETYPE/node_modules`...

**Comment by chaseadamsio** [comment]
Yeah, that sounds like a bug.

**Comment by ryan-roemer** [comment]
So far on NPM v2, I'm getting installs within `ARCHETYPE/node_modules` as is expected:

```
$ npm prune --production
unbuild eslint-config-defaults@7.1.1
unbuild eslint-plugin-filenames@0.1.2
unbuild eslint-plugin-react@3.8.0
unbuild karma-firefox-launcher@0.1.7
unbuild karma-ie-launcher@0.2.0
unbuild karma-phantomjs-launcher@0.2.1
unbuild karma-phantomjs-shim@1.1.2
unbuild karma-safari-launcher@0.1.1
unbuild karma-spec-reporter@0.0.20
unbuild karma-webpack@1.7.0
unbuild sinon@1.17.2
unbuild karma-chrome-launcher@0.2.1
unbuild react-hot-loader@1.3.0
unbuild karma-sauce-launcher@0.2.14
unbuild isparta-loader@1.0.0
unbuild karma-coverage@0.5.3
unbuild opener@1.4.1
unbuild mocha@2.3.3
unbuild phantomjs@1.9.18
unbuild webpack-dev-server@1.12.1
unbuild eslint@1.9.0
unbuild karma@0.13.15
unbuild karma-mocha@0.2.1
unbuild babel-eslint@4.1.5
(nvm  v0.10.40) rroemer@L-SB80X4FFT4-M ~/scm/fmd/formidable-react-component-boilerplate/node_modules/builder-react-component $ npm --version
2.13.5
```

@exogen -- Which NPM version are you using?

**Comment by exogen** [comment]
I'm on `3.3.10`.

**Comment by ryan-roemer** [comment]
@exogen -- Can you try with npm `v2.13.5` and report back here?

**Comment by exogen** [comment]
They're installed in the right place, under archetype. Being under the root is probably just npm 3 doing its little flattening trick. But without running `npm prune --production` like you did above, you still get all those errors from npm, which is what I was originally complaining about. When running `npm ls` from root:

```
npm ERR! extraneous: eslint-config-defaults@7.1.1 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/eslint-config-defaults
npm ERR! extraneous: eslint-plugin-filenames@0.1.2 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/eslint-plugin-filenames
npm ERR! extraneous: eslint-plugin-react@3.8.0 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/eslint-plugin-react
npm ERR! extraneous: karma-firefox-launcher@0.1.7 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-firefox-launcher
npm ERR! extraneous: karma-ie-launcher@0.2.0 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-ie-launcher
npm ERR! extraneous: karma-mocha@0.2.1 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-mocha
npm ERR! extraneous: karma-phantomjs-launcher@0.2.1 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-phantomjs-launcher
npm ERR! extraneous: karma-phantomjs-shim@1.1.2 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-phantomjs-shim
npm ERR! extraneous: karma-safari-launcher@0.1.1 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-safari-launcher
npm ERR! extraneous: opener@1.4.1 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/opener
npm ERR! extraneous: karma-spec-reporter@0.0.20 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-spec-reporter
npm ERR! extraneous: karma-webpack@1.7.0 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-webpack
npm ERR! extraneous: sinon@1.17.2 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/sinon
npm ERR! extraneous: karma-chrome-launcher@0.2.1 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-chrome-launcher
npm ERR! extraneous: mocha@2.3.3 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/mocha
npm ERR! extraneous: babel-eslint@4.1.5 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/babel-eslint
npm ERR! extraneous: phantomjs@1.9.18 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/phantomjs
npm ERR! extraneous: react-hot-loader@1.3.0 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/react-hot-loader
npm ERR! extraneous: webpack-dev-server@1.12.1 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/webpack-dev-server
npm ERR! extraneous: eslint@1.9.0 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/eslint
npm ERR! extraneous: isparta-loader@1.0.0 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/isparta-loader
npm ERR! extraneous: karma-sauce-launcher@0.2.14 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-sauce-launcher
npm ERR! extraneous: karma-coverage@0.5.3 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma-coverage
npm ERR! extraneous: karma@0.13.15 /Users/brianbeck/Projects/Formidable/Builder/victory-animation/node_modules/builder-react-component/node_modules/karma
```

Is there a way to make builder work without making npm spit out these errors?

**Comment by exogen** [comment]
To be clear, yes, I know why we need the archetype's dev deps and install them. I'm saying it's not gonna be worth using builder if you need to sacrifice sane npm output. :)

**Comment by ryan-roemer** [comment]
> I'm saying it's not gonna be worth using builder if you need to sacrifice sane npm output. :)

... uh, that's a totally separate discussion if we can't just "magically fix things" here.

**Comment by exogen** [comment]
Eh, that might be true if this were just an annoyance... I know those extraneously installed packages aren't technically hurting anything. The fact is that we break this:

```npm ls && echo OK```

...because npm does consider it a legitimate error. So really it's not just some annoying console output.

**Comment by ryan-roemer** [comment]
No one depends on `npm ls && echo OK` in real world usage. And hell, we can abstract this to `builder ls` if needbe. Important, and we should try to fix here -- but not critical, and easily bandaided if folks disagree with duplicated deps in root `package.json`.

**Comment by exogen** [comment]
OK, [check it](https://github.com/search?utf8=%E2%9C%93&q=%22npm+ls%22+language%3AJavaScript&type=Code&ref=searchresults). You don't think any of those will behave differently if the process is suddenly not returning exit code 0 anymore? And I am pretty sure I have personally seen `npm ls` in a build or CI script before.

**Comment by ryan-roemer** [comment]
@exogen -- I'm not saying folks "never do it" just that

1. It's not critical to normal workflows
2. When it is, can trivially be shell coded around `npm ls || true`
3. Or easily bandaided if we can't solve this instant issue here with adding deps to `package.json`

So, again, please focus on solving the problem -- I would appreciate your help. If we can't solve it with first class solution, we'll reevaluate .

**Comment by ryan-roemer** [comment]
Fixed in #18
